core/src/test/scala/org/apache/spark/util/JsonProtocolSuite.scala

```diff
@@ -934,6 +939,12 @@ private[spark] object JsonProtocolSuite extends Assertions {
     assert(newValue === expectedValue)
   }
 
+  private def testException(exception: Exception): Unit = {
+    val newException = JsonProtocol.exceptionFromJson(
+      toJsonString(JsonProtocol.exceptionToJson(exception, _)))
+    assertEquals(exception, newException)
+  }
```

Comment:
Thinking more, since we are adding this util method - we can remove `testStackTrace` method and delete to this as well. What we are trying to make sure is the exception deserializes to the same - for both the message and stack trace elements.